File: rust/leet_code/src/1544.rs
impl Solution {
    pub fn make_good(s: String) -> String {
        let mut stack = Vec::new();
        for c in s.chars() {
            if stack.is_empty() {
                stack.push(c);
            } else {
                let top = stack.last().unwrap();
                if top.is_lowercase() && top.to_ascii_uppercase() == c {
                    stack.pop();
                } else if top.is_uppercase() && top.to_ascii_lowercase() == c {
                    stack.pop();
                } else {
                    stack.push(c);
                }
            }
        }
        stack.into_iter().collect()
    }
}

File: swift/leet_code/1544.swift
class Solution {
    func makeGood(_ s: String) -> String {
        var stack = [Character]()
        for c in s {
            if stack.isEmpty {
                stack.append(c)
            } else {
                let last = stack.last!
                if last.isLowercase, last.uppercased() == String(c) {
                    stack.removeLast()
                } else if last.isUppercase, last.lowercased() == String(c) {
                    stack.removeLast()
                } else {
                    stack.append(c)
                }
            }
        }
        return String(stack)
    }
}
